### PR TITLE
Add Data over time for charts in ListSkillService #1317

### LINKED ISSUE #1317: Add Data over time for charts in ListSkillService
**Actual Behaviour**

Not present

**Expected Behaviour**

For PR https://github.com/fossasia/susi.ai/issues/2550, we require time data for endpoints

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by fragm3**
Fixes #1317

Changes: 

- [x] Logins over time
- [x] User registration over time
- [x] Skill edits over time
- [x] Skill creations over time
- [x] Skill accessed over time

### CODE REVIEW

**Comment by fragm3** [comment]
For API: http://127.0.0.1:4000/aaa/getUsers.json?access_token=ljpJ0is16JxiHCuKUq6kitfO0hpCqg&getUserStats=true

Getting response:

```
{
  "accepted": true,
  "lastLoginTime": {
    "count": 1,
    "timestamp": "Wed, 17 Jul 2019 18:52:19 +0000"
  },
  "signupTime": {
    "count": 1,
    "timestamp": "Wed, 17 Jul 2019 17:09:37 +0000"
  },
  "userStats": {
    "anonymous": 0,
    "users": 7,
    "reviewers": 0,
    "operators": 0,
    "admins": 0,
    "superAdmins": 1,
    "activeUsers": 8,
    "inactiveUsers": 0,
    "totalUsers": 8
  },
  "message": "Success: Fetched all users stats!",
  "session": {"identity": {
    "type": "host",
    "name": "127.0.0.1_ba69b956",
    "anonymous": true
  }}
}
```

**Comment by fragm3** [comment]
@akshatnitd @Orbiter, please review

File: src/ai/susi/server/api/cms/CreateSkillService.java [hunk 1]
```diff
@@ -61,7 +61,7 @@
 
     @Override
     public UserRole getMinimalUserRole() {
-        return UserRole.USER;
```
Comment (akshatnitd) [inline src/ai/susi/server/api/cms/CreateSkillService.java]: Why this change?

Comment (fragm3) [reply src/ai/susi/server/api/cms/CreateSkillService.java]: Oh, sorry left it by mistake

**Comment by fragm3** [comment]
@akshatnitd I have updated my comment. About the skill data over time- I have to test it on real data to be sure, it works. If it requires changes, I'll create a new PR.

**Comment by fragm3** [comment]
@akshatnitd yup, correct. Made the changes, updated response:

```
{
  "accepted": true,
  "lastLoginOverTime": [{
    "timeStamp": "Jul 2019",
    "count": 5
  }],
  "signupOverTime": [{
    "timeStamp": "Jul 2019",
    "count": 9
  }],
  "userStats": {
    "anonymous": 0,
    "users": 10,
    "reviewers": 0,
    "operators": 0,
    "admins": 0,
    "superAdmins": 1,
    "activeUsers": 11,
    "inactiveUsers": 0,
    "totalUsers": 11
  },
  "message": "Success: Fetched all users stats!",
  "session": {"identity": {
    "type": "email",
    "name": "shubh12345am@gmail.com",
    "anonymous": false
  }}
}
```

File: src/ai/susi/server/api/aaa/GetUsers.java [hunk 2]
```diff
@@ -36,7 +39,7 @@ public String getAPIPath() {
 
     @Override
     public UserRole getMinimalUserRole() {
-        return UserRole.OPERATOR;
+        return UserRole.USER;
```
Comment (akshatnitd) [inline src/ai/susi/server/api/aaa/GetUsers.java]: USER?

**Comment by fragm3** [comment]
@akshatnitd fixed
